File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    dummy := &ListNode{}
    out := dummy
    firstList := list1
    secondList := list2

    if firstList == nil {
        out = secondList
        return out
    } else if secondList == nil {
        out = firstList
        return out
    }

    for firstList != nil && secondList != nil {
        if firstList.Val <= secondList.Val {
            out.Next = firstList
            firstList = firstList.Next
            out = out.Next
        } else {
            out.Next = secondList
            secondList = secondList.Next
            out = out.Next
        }
        if firstList == nil {
            out.Next = secondList
        } else if secondList == nil {
            out.Next = firstList
        }
    }

    return dummy.Next
}

File: Data Structures & Algorithms/add-two-numbers/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        #l1 number
        l1num = ""
        while l1:
            l1num = str(l1.val) + l1num
            l1 = l1.next

        #l2 number
        l2num = ""
        while l2:
            l2num = str(l2.val) + l2num
            l2 = l2.next

        l3num = str(int(l1num) + int(l2num))
        l3 = ListNode()
        res = l3
        for i in range(len(l3num)-1, -1, -1):
            tmp = ListNode(int(l3num[i]))
            l3.next = tmp
            l3 = l3.next

        return res.next
